File: dotnet_and_angular/ProductStore/Server/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace Server.Models;

public class Product {
    [Key]
    public int Id { get; set; }

    [Required]
    public required string Name { get; set; }

    public string? Description { get; set; }

    [Required]
    public decimal Price { get; set; }

    public bool IsAvailable { get; set; }
}

File: dotnet_and_angular/ModuleAndInterceptor/Client/src/app/product-module/models/product.ts
/**
 * The interface is a purely compile-time construct for type-hinting
 * and conformity in Typescript. It doesn’t translate to a concrete
 * JavaScript class during runtime.
 */
export interface Product {
  id: number;
  name: string;
  description: string | null;
  price: number;
  isAvailable: boolean;
}
